src/Service/SF1500/BrugerService.php

```diff
@@ -49,80 +56,153 @@ public function list(array $ids, array $fields = []): array
+                    foreach ($items as $item) {
+                        $adresseId = $item->getRelation(Bruger::RELATION_ADRESSE, $field);
+                        if (isset($adresser[$adresseId])) {
+                            $item->{$fieldMap[$field]} = $adresser[$adresseId]->adressetekst;
+                        }
+                    }
+                }
             }
         }
 
-        return $model;
+        return $items;
     }
 
-    protected function doSoeg(array $query): SoegOutputType
+    /**
+     * {@inheritdoc}
+     */
+    protected function doSoeg(array $query): ?SoegOutputType
     {
         $attributListe = new AttributListeType();
-        if (isset($query['brugernavn'])) {
+        if (isset($query[self::FILTER_BRUGERNAVN])) {
             $attributListe->addToEgenskab((new EgenskabType())
-                ->setBrugerNavn($query['brugernavn']));
+                ->setBrugerNavn($query[self::FILTER_BRUGERNAVN]));
         }
 
         $relationListe = new RelationListeType();
-        if (isset($query['email'])) {
+        if (isset($query[self::FILTER_EMAIL])) {
             // TODO We cannot search by email yet
         }
+        if (isset($query[self::FILTER_PERSON_ID])) {
+            $ids = (array)$query[self::FILTER_PERSON_ID];
+            foreach ($ids as $id) {
+                $relationListe->addToTilknyttedePersoner((new PersonFlerRelationType())
+                    ->setReferenceID(new UnikIdType($id)));
+            }
+        }
 
         $request = (new SoegInputType())
+            ->setMaksimalAntalKvantitet((int)($query['limit'] ?? self::DEFAULT_LIMIT))
+            ->setFoersteResultatReference((int)($query['offset'] ?? 0))
             ->setAttributListe($attributListe)
             ->setRelationListe($relationListe);
 
-        return $this->clientSoeg()->soeg($request);
+        $response = $this->clientSoeg()->soeg($request) ?: null;
+
+        if (null !== $response && isset($query[self::FILTER_LEDER])) {
+            // Keep or throw away managers.
+            $managerIds = $this->getManagerIds();
+
+            $brugerIds = $response->getIdListe()->getUUIDIdentifikator();
+            $response->setIdListe(new IdListeType(
+                array_values(
+                    $query[self::FILTER_LEDER]
+                        // Keep only managers.
+                        ? array_intersect($brugerIds, $managerIds)
+                        // Throw away managers.
+                        : array_diff($brugerIds, $managerIds)
+                )
+            ));
+        }
+
+        return $response;
+    }
+
+    private function getManagerIds()
+    {
+        $service = $this->getService(OrganisationFunktionService::class);
+        /** @var OrganisationFunktion[] $organisationFunktioner */
+        $organisationFunktioner = $service->soeg([
+            OrganisationFunktionService::FILTER_FUNKTIONSTYPEID => $this->options['organisation-funktion-manager-id']
+        ]);
```

Comment:
Could we end up in a scenario where this list is not exhaustive?

Reply:
I'm not sure I understand this question. We want to search for a specific type (id) and therefore search, i.e. filter, only on `OrganisationFunktionService::FILTER_FUNKTIONSTYPEID`.